File: Activity/Week3/C/leftmost-column-with-at-least-a-one.c
/**
 * // This is the BinaryMatrix's API interface.
 * // You should not implement it, or speculate about its implementation
 * struct BinaryMatrix {
 *     int (*get)(struct BinaryMatrix*, int, int);
 *     int* (*dimensions)(struct BinaryMatrix*);
 * };
 */

int leftMostColumnWithOne(struct BinaryMatrix* matrix) {
    int row = matrix->dimensions(matrix)[0];
    int col = matrix->dimensions(matrix)[1];
    int leftMost = col;
    
    for(int i = 0; i < row;i++)
    {
        while(matrix->get(matrix, i, leftMost - 1) == 1)
        {
            leftMost--;
            
            if(leftMost == 0)
                return 0;
        }
    }
    
    if(leftMost == col)
        return -1;
    
    return leftMost;
}

File: Activity/Week3/Python/leftmost-column-with-at-least-a-one.py
# """
# This is BinaryMatrix's API interface.
# You should not implement it, or speculate about its implementation
# """
#class BinaryMatrix(object):
#    def get(self, x: int, y: int) -> int:
#    def dimensions(self) -> list[]:

class Solution:
    def leftMostColumnWithOne(self, binaryMatrix: 'BinaryMatrix') -> int:
        row, col = binaryMatrix.dimensions()
        leftMost = col
        
        for i in range(row):
            while binaryMatrix.get(i, leftMost - 1) == 1:
                leftMost -= 1
                
                if leftMost == 0:
                    return 0
                
        if leftMost == col:
            return -1
        
        return leftMost
